third_party/move/move-compiler-v2/tests/ability-transform/by_reference.exp

```diff
@@ -3787,74 +4555,108 @@ script {
 
 
 main() /* def_idx: 0 */ {
-L0:	$t13: u64
-L1:	c: &mut u64
-L2:	$t16: vector<u8>
-L3:	b: &mut vector<u8>
+L0:	$t4: vector<u8>
+L1:	$t6: vector<u8>
+L2:	$t14: vector<u8>
+L3:	$t16: vector<u8>
+L4:	$t19: u64
+L5:	c: &mut u64
+L6:	$t22: vector<u8>
+L7:	b: &mut vector<u8>
+L8:	$t30: vector<u8>
+L9:	$t32: vector<u8>
+L10:	$t38: vector<u8>
+L11:	$t40: vector<u8>
 B0:
-	0: LdU64(0)
-	1: LdU64(0)
-	2: Eq
-	3: BrFalse(45)
-B1:
-	4: LdConst[0](Vector(U8): [5, 104, 101, 108, 108, 111])
-	5: LdConst[0](Vector(U8): [5, 104, 101, 108, 108, 111])
+	0: LdConst[0](Vector(U8): [5, 104, 101, 108, 108, 111])
```

Comment:
It is very concerning that this optimizing rewrite is creating `75%` more code.

This is because `==` does not always create a copy (hence the lint `third_party/move/tools/move-linter/src/stackless_bytecode_lints/avoid_copy_on_identity_comparison.rs` checks for liveness information).